mod/01_rating/set/type/metric_value.rb

```diff
@@ -247,59 +245,65 @@ def legend args
 
   view :timeline_data do |args|
     # container elements
-    dot =  content_tag(:div, '', class: 'timeline-dot')
-    value_details_toggle = content_tag(
-                                        :i,
-                                        '',
-                                        class: 'fa fa-caret-right
-                                                fa-lg margin-left-10
-                                                btn btn-default btn-sm',
-                                        data: { toggle: 'collapse-next',
-                                                parent: '.value',
-                                                collapse: '.metric-value-details'
-                                              }
-                                      )
-    credit = wrap_with :div, class: 'credit' do
+    value_details = _render_value_details(args)
+    year = content_tag(:span, card.cardname.right)
+    year << content_tag(:div, '', class: 'timeline-dot')
+    year = content_tag(:div, year.html_safe, class: 'td year')
+    value =  _render_modal_details(args)
+    value << content_tag(:span, legend(args), class: 'metric-unit')
+    value << _render_value_details_toggle
+    value << value_details.html_safe
```

Comment:
good improvements but you can go one step further and extract views  for `year` and `value`